### ISSUE TITLE
simplify, factor, etc do not work on Matrix

### ISSUE THREAD

**Issue by HDembinski** [sympy/sympy#22569]
Expected behavior:
Matrix should behave like an expression, having methods `.simplify()`, `.factor()`, etc. Calling this should apply the method on each element of the matrix.

Current behavior:
Calling `Matrix.simplify()` etc. returns `None` instead of the simplified Matrix, and the simplification is not applied element-wise.

**Comment by ThePauliPrinciple** [comment]
This can currently be done with `M.applyfunc(lambda e: e.simplify())`. Note that `Matrix` is not a subclass of `Expr`, so should not be expected to behave the same way as an `Expr`. It would be convenient to have wrapper methods for applying `simplify` element wise, although I'm not sure it should be called `.simplify`.

**Comment by ThePauliPrinciple** [comment]
If you want to have a `Matrix` which behaves like an `Expr`, use `ImmutableMatrix`:
```python
>>> x = Symbol('x')
>>> ImmutableMatrix([cos(x)**2 + sin(x)**2]).simplify()
[1]
```

**Comment by oscargus** [comment]
But it is a bit weird that `None` is returned when calling `simplify`...

```
In [35]: (cos(x)**2 + sin(x)**2)*Matrix([[1, 1], [2, 2]])
Out[35]: 
⎡     2         2            2         2     ⎤
⎢  sin (x) + cos (x)      sin (x) + cos (x)  ⎥
⎢                                            ⎥
⎢     2           2          2           2   ⎥
⎣2⋅sin (x) + 2⋅cos (x)  2⋅sin (x) + 2⋅cos (x)⎦

In [36]: ((cos(x)**2 + sin(x)**2)*Matrix([[1, 1], [2, 2]])).simplify()

In [37]: ((cos(x)**2 + sin(x)**2)*ImmutableMatrix([[1, 1], [2, 2]])).simplify()
Out[37]: 
⎡1  1⎤
⎢    ⎥
⎣2  2⎦
```

**Comment by ThePauliPrinciple** [comment]
Actually, this makes perfect sense. It isn't a `Basic`, so instead of returning a new `Basic` object, it actually just applies it to the current object

**Comment by oscargus** [comment]
I see your point from a class design perspective, but not really from a user perspective. Especially combined with the `ImmutableMatrix` behavior (where, of course, it is not an option to modify the object). But I guess it is a design decision.

**Comment by oscarbenjamin** [comment]
It is unfortunate that mutable matrices behave this way but not easy to change given the compatibility issues.

**Comment by HDembinski** [comment]
> I see your point from a class design perspective, but not really from a user perspective. Especially combined with the `ImmutableMatrix` behavior (where, of course, it is not an option to modify the object). But I guess it is a design decision.

I fully agree. The API of sympy should be user friendly, since non-experts are working with it directly. In that case, the API should be very consistent from the user perspective.

Why is it a compatibility issue to change behavior? I don't see that. Right now, Matrix.simplify does not do anything useful, so I doubt that changing it will break anyones code. As nobody uses this.

**Comment by oscarbenjamin** [comment]
> Right now, Matrix.simplify does not do anything useful, so I doubt that changing it will break anyones code. As nobody uses this.

It definitely is useful and it used and changing it would break someone's code.
